src/features/feedback/components/FeedbackForm.vue

```diff
@@ -0,0 +1,191 @@
+        >Wat is de aanleiding?</label
+      >
+      <textarea
+        id="aanleiding"
+        v-model="feedback.aanleiding"
+        class="utrecht-textarea utrecht-textarea--html-textarea"
+      ></textarea>
+
+      <label for="contactgegevens" class="utrecht-form-label"
+        >Het emailadres of telefoonnummer waarop de auteur jou kan
+        bereiken</label
+      >
+      <input
+        type="text"
+        id="contactgegevens"
+        v-model="feedback.contactgegevens"
+        class="utrecht-textbox utrecht-textbox--html-input"
+      />
+    </fieldset>
+
+    <application-message
+      class="formValidationMessage"
+      v-if="validationMessage != ''"
+      messageType="error"
+      :message="validationMessage"
+    ></application-message>
+
+    <application-message
+      v-if="serviceResult.error"
+      messageType="error"
+      message="Er is een fout opgetreden"
+    ></application-message>
+
+    <menu>
+      <button
+        @click="cancelDialog.reveal"
+        class="utrecht-button utrecht-button--secondary-action"
+        type="button"
+      >
+        Annuleren
+      </button>
+
+      <button class="utrecht-button utrecht-button--submit" type="submit">
+        Opslaan
+      </button>
+    </menu>
+  </form>
+
+  <!-- Annuleer Dialog -->
+  <modal-template v-if="cancelDialogRevealed">
+    <template #message>
+      <paragraph>
+        Weet u zeker dat u wilt annuleren? Alle gegevens worden verwijderd.
+      </paragraph>
+    </template>
+
+    <template #menu>
+      <button
+        @click="cancelDialog.cancel"
+        class="utrecht-button utrecht-button--secondary-action"
+      >
+        Nee
+      </button>
+      <button @click="cancelDialog.confirm" class="utrecht-button">Ja</button>
+    </template>
+  </modal-template>
+</template>
+
+<script lang="ts" setup>
+import { ref, reactive, defineProps } from "vue";
+import { useFeedbackService } from "../service";
+import type { Feedback, ServiceResult } from "../types";
+import { useConfirmDialog } from "@vueuse/core";
+import SimpleSpinner from "@/components/SimpleSpinner.vue";
+import ApplicationMessage from "@/components/ApplicationMessage.vue";
+import ModalTemplate from "@/components/ModalTemplate.vue";
+import Paragraph from "@/nl-design-system/components/Paragraph.vue";
+
+const props = defineProps<{
+  id: unknown | URL;
+  name: string;
+}>();
+
+const serviceResult = ref<ServiceResult>({} as ServiceResult);
+const service = useFeedbackService();
+const cancelDialogRevealed = ref(false);
+const cancelDialog = useConfirmDialog(cancelDialogRevealed);
+const emit = defineEmits(["cancelled", "saved"]);
+const feedback: Feedback = reactive({
+  naam: props.name,
+  uri: props.id,
+  content: "",
+  opmerking: "",
+  aanleiding: "",
+  contactgegevens: "",
+});
+
+const validationMessage = ref("");
+
+cancelDialog.onConfirm(() => annuleren());
+
+//validate
+//opslaan
+const submit = () => {
+  validationMessage.value = feedback.opmerking
```

Comment:
Het lijkt me netter en minder bewerkelijk om van html validatie gebruik te maken (required attribuut). Heb je daar bewust niet voor gekozen?

Reply:
aangepast in overleg